### DISCUSSION TITLE
Rotated ComboBox Popup Placement

### DISCUSSION THREAD

**Discussion by collinmdp** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#2934] [Q&A] [answered]
I am trying to rotate a ComboBox. The ComboBox and all the items in the popup rotate correctly. However, the popup is in the wrong location. How do I change the position of the Popup so that it appears to the right of the ComboBox?

When the `Width` is 75, the popup is 150 lower.
![image](https://user-images.githubusercontent.com/37085627/200071317-a46d8411-bc50-4582-8609-858bbb85c2dd.png)

When the `Width` is 150, the popup is 150 lower.
![image](https://user-images.githubusercontent.com/37085627/200070493-e4bdd24f-b759-486c-bb73-7f3eaf6cc123.png)

```
<ComboBox Style="{StaticResource MaterialDesignFloatingHintComboBox}" Width="150"
          Grid.Row="1" ItemsSource="{Binding Path=Configuration.Items}"
          SelectedItem="{Binding Path=Configuration.SelectedItem}"
          materialDesign:HintAssist.Hint="Selected Item" Margin="0 5">
    <ComboBox.LayoutTransform>
        <RotateTransform Angle="-90"/>
    </ComboBox.LayoutTransform>
    <ComboBox.ItemsPanel>
        <ItemsPanelTemplate>
            <StackPanel Orientation="Vertical">
                <StackPanel.LayoutTransform>
                    <RotateTransform Angle="90"/>
                </StackPanel.LayoutTransform>
            </StackPanel>
        </ItemsPanelTemplate>
    </ComboBox.ItemsPanel>
</ComboBox>
```

**Comment by nicolaihenriksen** [comment]
@collinmdp I believe this is just a bug. Rotating a standard `ComboBox` yields this result:

![image](https://user-images.githubusercontent.com/19572699/209099909-80285a11-cc7e-44a6-a84e-7fe2ae0f3e3c.png)

Simply applying the `MaterialDesignFloatingHintComboBox` style (or any of the MDIX `ComboBox` styles) to it yields this result:
![image](https://user-images.githubusercontent.com/19572699/209100081-2103dc28-5535-43d0-aa00-0bc701c0cb82.png)

I'll create an issue for this and look into fixing it...

**Reply by nicolaihenriksen** [reply]
I have created issue #3004 for this. I'll see if I can fix it...

**Comment by nicolaihenriksen** [comment, marked as answer]
@collinmdp The latest release, 4.7.0, has support for rotation of a `ComboBox`. However, the rotation is something that needs to be handled by the consumer via a callback. You can see examples of +- 90 degrees rotation in the demo app.

**Comment by collinmdp** [comment]
Thank you for the help
